auth/resource.cc

```diff
@@ -50,13 +53,27 @@ static permission_set applicable_permissions(const data_resource_view& dv) {
                 permission::AUTHORIZE>();
     }
 
-    return permission_set::of<
+    // keyspace
+    if (dv.keyspace()) {
+        return permission_set::of<
             permission::CREATE,
             permission::ALTER,
             permission::DROP,
             permission::SELECT,
             permission::MODIFY,
             permission::AUTHORIZE>();
+    }
+
+    // All keyspaces
```

Comment:
We need more information here, why this split dv.keyspace vs all keyspaces.

Also you can probably use add() to permission_set to keep some common code